sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -128,6 +130,76 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
         ShimLoader.setSparkShimProviderClass(conf.shimsProviderOverride.get)
       }
 
+      try {
+        // Compare if the cudf version mentioned in the classpath is equal to the version which
+        // plugin expects. If there is a version mismatch, throw error. This check can be disabled
+        // by setting this config spark.rapids.cudfVersionOverride=true
+        val cudfPropertiesFileName = "cudf-java-version-info.properties"
+        val pluginPropertiesFileName = "rapids4spark-version-info.properties"
+
+        val props = new Properties
+        val cudfClassLoader = classOf[ai.rapids.cudf.ColumnVector].getClassLoader
+        val cudfProperties = cudfClassLoader.getResourceAsStream(cudfPropertiesFileName)
+        if (cudfProperties == null) {
+          val errorMsg = s"Could not find properties file $cudfPropertiesFileName in the cudf " +
+            "jar. Cannot verify cudf version compatibility with RAPIDS Accelerator version."
+          if (!conf.cudfVersionOverride) {
+            throw new RuntimeException(errorMsg)
```

Comment:
Don't throw two types of exceptions, only the custom one that's derived from `RuntimeException`.  Once we do that, we no longer need to calculate the message in a local since there's only one thing to do with it.